### PR TITLE
Add Markdown component

### PR DESCRIPTION
**Pull request by jjt**
Defines a <Markdown/> component that uses mdast to target
spectacle components.

Uses [my fork of mdast-react](https://github.com/jjt/mdast-react) until mapbox/mdast-react#9 lands.

**Todo**
- [x] Blockquotes
- [x] Expose mdast config to user
- [x] Inline code
- [ ] Update README
- [ ] Tweak markdown example slides (if necessary)

### CODE REVIEW

File: src/markdown.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  return React.createClass({
+    render(){
+      const props = {...this.props, ...boundProps};
+      return React.createElement(component, {...props}, this.props.children);
+    }
+  });
+}
+
+// Would like to expose this config to the user
+const mdastConfig = {
```
Comment (ryan-roemer) [inline src/markdown.jsx]: Could we just make this a property on `Markdown` to allow consumers to override the class if they want?

File: src/markdown.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    p: Text,
+    strong: spectacleComponent(S, {type: "bold"}),
+    ul: List
+  }
+};
+
+
+export default class Markdown extends React.Component {
+  render() {
+    const content = typeof this.props.source !== "undefined" ?
```
Comment (ryan-roemer) [inline src/markdown.jsx]: You've got a _different_ check for this same thing at https://github.com/FormidableLabs/spectacle/pull/87/files#diff-2d8f174e85e5217d3672c29687ab543cR12 -- maybe harmonize them?

**Comment by ryan-roemer** [comment]
@jjt -- Love it!!!

File: src/markdown.jsx (hunk 2 above)
Comment (jjt) [inline src/markdown.jsx]: Yup, good call.

File: src/markdown.jsx (hunk 1 above)
Comment (jjt) [inline src/markdown.jsx]: Oooh, nice, I like it.

File: src/markdown.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        {mdast().use(mdastReact, mdastConfig).process(content)}
+      </div>
+    );
+  }
+}
+
+Markdown.propTypes = {
+  children: React.PropTypes.node,
+  source: React.PropTypes.string
+}
```
Comment (jjt) [inline src/markdown.jsx]: Also need `mdastConfig` - I'll add that rn.

**Comment by kenwheeler** [comment]
@jjt Awesome! let me know when its good to go

**Comment by jjt** [comment]
Previous PR to mdast-react landed, submitted a new one to support rendering blockquote content without a wrapping `<p>` element. Targeting my fork once again https://github.com/jjt/mdast-react/commit/a6d6b02

I noticed that there's no support for inline code blocks, so tackling that next.

Then we should be GTG.

**Comment by jjt** [comment]
Looks like rendering inline code in `<Text>` components is a little wonky in Spectacle currently.

```markdown
> Choose from `inline code` or code blocks:
```

When `inlineCode: Code` is passed in as an option, mdast-react turns the above into
```js
<Text>Choose from <Code>inline code</Code> or code blocks:</Text>
```
Which looks like this:

image

But if we don't specify `inlineCode: Code` in the options to mdast-react we get something that looks okay, albeit unstyled.

image

**Comment by jjt** [comment]
Alright, support for inline code is sketched out but due to https://github.com/FormidableLabs/spectacle/issues/88 we should wait on passing `inlineCode: Code` to mdast-react.

**Comment by jjt** [comment]
@kenwheeler I think the code itself is GTG. We can hold off until those PRs land in mdast-react, or we can use my fork.

I'll need to get your opinion on two things (or just hand these tasks over to you):

Example Markdown Slide content
README additions

**Comment by kenwheeler** [comment]
@jjt I'll add that stuff and then get this merged

**Comment by jjt** [comment]
@kenwheeler Great, let me know if I can help in any way.

**Comment by kenwheeler** [comment]
@jjt I'm getting the 0.14 conversion PR in today, after that did you want to update to 0.14 compat or do you want me to during the merge?

**Comment by kenwheeler** [comment]
Ok @jjt I have this all set locally, just messing around with the slide design & documentation. Merging soon.

**Comment by jjt** [comment]
:confetti_ball:

**Pull request merged**